packages/core/docs/advanced/server-middleware.md

```diff
@@ -0,0 +1,138 @@
+# Server middleware
+
+The server middleware is a part of Vue Storefront networking. That's the way of making connections with the eCommerce platform and also a solution to reduce bundle size, number of calls, credentials sharing, extensibility handling, and more.
+
+## What is Vue Storefront Middleware and why we need it?
+
+The Vue Storefront middleware is an express.js proxy that taking the requests from the front-end translates them to a certain integration and call-related API-client.
+
+We have implemented it for a variety of reasons.
+
+First of all, it allows us to provide a prover way of extensibility - As a developer, you have control of the requests and the responses in the given platform and you can write an extension to add something extra in your project.
+
+A front-end application sometimes needs an additional API endpoint - that's also possible since you have a server written in express.js.
+
+All credentials for your platform are stored only on the backend side, there are no shared keys in the browser.
+
+## How it works (in a nutshell)
```

Comment:
this part is really hard to understand, even with a prior knowledge. I suppose it will be much harder for newcomers.